### DISCUSSION TITLE
Deploy Renovate on Openshift

### DISCUSSION THREAD

**Discussion by itpjob** [renovatebot/renovate#13503] [Request Help]
Hi,

We have relevant helmfile, config cron job and the secret config file ..The cron job gets added in OpenShift Resources. However, it does not start any pod or anything. Is there anything else that we need to take care of? Do we need to deploy (using a deployment.yaml) renovate in a pod to get the cron job fired?

apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: renovate
spec:
  schedule: "* * * * *"
  concurrencyPolicy: Forbid
  jobTemplate:
    spec:
      template:
        spec:
          containers:
            - name: renovate
              # Update this to the latest available and then enable Renovate on
              # the manifest
              image: "our organisation registry"
              args:
                - "our specific repository"
              # Environment Variables
              env:
                - name: LOG_LEVEL
                  value: debug
              envFrom:
                - secretRef:
                    name: renovate-env
          restartPolicy: Never

Secret file looks like below:
apiVersion: v1
kind: Secret
metadata:
  name: renovate
type: Opaque
stringData:
  RENOVATE_AUTODISCOVER: 'false'
  RENOVATE_PLATFORM: 'gitlab'
  RENOVATE_TOKEN: 'secret token'

the chart.yaml is below:
apiVersion: v2
description: Renovate minimal chart
name: renovatebot
version: 1.0.2

helmfile is similar to below ( i had to pull out any sensitive entries)

helmDefaults:
  atomic: true
  tls: false
  wait: true
  timeout: 600
  createNamespace: false

bases:
  - ./environments.yaml

values:
  - appName: {{ env "APP_NAME" | default "XXX" }}
    name: xx-renovatebot-{{ .Environment.Name }}
    kubeDomain: "our kube domain"
    image: "our image registry"

---
releases:
  - name: {{ .Values.name }}
    namespace: {{ .Namespace }}
    chart: renovatebot-chart
    version: 1.0.0
    values:
      - name: {{ .Values.name }}
        image: {{ .Values.image }}
        applicationDomain: {{ .Values.name }}.{{ .Values.kubeDomain }}
        imagePullPolicy: Always

**Comment by sethjones** [comment]
Hi,
I have been able to host it on my corporate openshift without much hassle.

If you use the renovate helm chart hosted [here](https://artifacthub.io/packages/helm/renovate/renovate), you can override many of the chart values in your own values file.  Specifically, disabling the podSecurityContext and containerSecurityContext was required for me.  

Other specifics of your environment might impact it as well.  IE, I have to specifically define cpu/memory resources in our environment.

**Reply by rarkins** [reply]
@sethjones do you think it's possible to have a universal chart for Openshift and non, or is it more common to publish separate charts?

**Reply by sethjones** [reply]
Yes, you can use the common chart and just override the values. This is how its intended to be used.  The values are used for customization.
I didnt have to update the chart, just adjust the values.

**Reply by rarkins** [reply]
Thanks, I misunderstood
